learningequality/studio
# Remove API request consolidation across tabs.

## Summary
### Description of the change(s) you made
* In an attempt to limit duplicate requests to the server, there was a slightly overcomplicated system using broadcast channel to limit API requests using the resource layer to only originate from one tab
* This relied on a vendored, edited copy of the broadcast-channel leaderElection functionality, in order to be able to defer all data loading until after a leader had been elected
* Unfortunately this vendored version caused an issue where every tab became a leader, so nothing was solved
* This issue was fixed in https://github.com/learningequality/studio/pull/3251 by reverting to an upstream version of broadcast-channel's leaderElection
* However, because we are no longer waiting for the leaderElection to be decided, requests can be initiated before a leader exists, and so are not responded to and requested from the server
* This PR resolves this issue by removing this functionality completely and just initiating API requests from each tab individually
* In the future, similar functionality can be achieved in a more robust way by using Service Worker request interception.

### Manual verification steps performed
1. Clear indexedDB
2. Load any page in Studio
3. Observe that data fetches complete

## Reviewer guidance
### How can a reviewer test these changes?

Are there any extant references to `requestCollection` or `requestModel` that I missed?
Do pages load as expected.


### Are there any risky areas that deserve extra testing?
I think if anything, this reduces complexity and risk for the most part.

----

### Contributor's Checklist


PR process:


Studio-specifc:



Testing:

___

### Reviewer's Checklist
#### This section is for reviewers to fill out.
